docs/proposals/11-cluster-health.md

```diff
@@ -0,0 +1,205 @@
+If there is more specific information about the issue then more specific condition types can be defined.
+
+Such conditions could be updated manually by the administrators, operators on duty based on publications of such issues by the cloud providers.
+If there is a way to automatically scrape such information from well-known locations or APIs then this could even be automated.
+
+##### Possible Actions
+
+- Alerts can be raised to the administrator of the cluster as well as the operator/support team on duty.
+- Controllers such as `kube-controller-manager`, `machine-controller-manager` that manage crucial resources such as `nodes` can be either scaled down as describe [above](#possible-actions) or enhanced to change their behaviour to ramp down their active interventions to the cluster.
+- Some resource creations (say, `pods`) could be throttled/rate-limited.
+- Gardener can defer any disruptive updates until later.
+
+#### CloudProviderQuotaExceeded
+
+If there is a way to detect that some cloud provider quotas are exceeded this information can be updated in the conditions with the `CloudProviderQuotaExceeded` condition type.
+If more specific information is available about which particular quota is exceeded then more specific condition types can be defined.
+
+##### Possible Actions
+
+- Alerts can be raised to the administrator of the cluster as well as the operator/support team on duty.
+- Some resource creations (say, `pods` or `LoadBalancer` `services`) could be throttled/rate-limited.
+
+#### ScaleThresholdReached
+
+If the cluster grows beyond a threshold (a threshold beyond which the cluster health may not be reliably guaranteed), this information can be updated in the conditions with `ScaleThresholdReached` condition type.
+If more information is available about which aspect of the cluster cannot reliably scale anymore (e.g. the control-plane, nodes, loadbalancers etc.) then more specific condition types can be defined.
+
+##### Possible Actions
+
+- Alerts can be raised to the administrator of the cluster as well as the operator/support team on duty.
+- Some resource creations (say, `pods` or `LoadBalancer` `services`) could be throttled/rate-limited.
+- Even the request rate to the control-plane could be throttled/rate-limited in an intelligent way to so that critical control-plane and system components get the throughput they need but other non-critical components get throttled/rate-limited.
+- Gardener can defer any disruptive updates until later or ramp them down.
+
+## Alternative
+
+Instead of re-using the [`Cluster`](https://github.com/kubernetes/cluster-registry/blob/master/pkg/apis/clusterregistry/v1alpha1/types.go#L30) resource from the [cluster-registry](https://github.com/kubernetes/cluster-registry), a new [`CustomResourceDefinition`](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#customresourcedefinitions) could be defined for this purpose.
```

Comment:
Why the Shoot resource from the Gardener API, or the Cluster resource CRD in the seed are not used?

Reply:
> Why the Shoot resource from the Gardener API, or the Cluster resource CRD in the seed are not used?

I have explained [here](https://github.com/gardener/gardener/pull/1797/files#r367814260) and [here](https://github.com/gardener/gardener/pull/1797/files#r367827634). I will update the document as well.